hypervisor/lib/string.c

```diff
@@ -439,28 +440,26 @@ char *strstr_s(const char *str1, size_t maxlen1, const char *str2, size_t maxlen
 	len1 = strnlen_s(str1, maxlen1);
 	len2 = strnlen_s(str2, maxlen2);
 
-	if (len1 == 0U) {
+	if (len1 < len2) {
 		return NULL;
 	}
 
-	/*
-	 * str2 points to a string with zero length, or
-	 * str2 equals str1, return str1
-	 */
-	if (len2 == 0U || str1 == str2) {
+	/* return str1 if str2 equals to str1 or str2 points to a string with zero length*/
+	if ((str1 == str2) || (len2 == 0U)) {
 		return (char *)str1;
 	}
 
+	str = str1;
 	while (len1 >= len2) {
 		for (i = 0U; i < len2; i++) {
-			if (str1[i] != str2[i]) {
+			if (str[i] != str2[i]) {
```

Comment:
we already have str1/str2, adding "str" is ugly... how about pstr, or str0 ?

Reply:
str1 -> haystack
str2 -> needle ?

Reply:
What does this mean? too complicated?